src/UI/Dice/DicePlay.razor

```diff
@@ -0,0 +1,185 @@
+                            </td>
+                        }
+                    }
+                    else
+                    {
+                        @for (var c = DiceEngine.BoardSize - 1; c >= 0; c--)
+                        {
+                            var col = c;
+                            var cell = board[row, col];
+                            <td @key=@((row, col))>
+                                <table style="margin: 1px; min-width: 50px; min-height: 50px; background: @GetCellBackground(board.GetCellIndex(row, col)); border-radius: 3px;">
+                                    <tbody>
+                                    <tr>
+                                        @for (int i = 0; i < 2; i++)
+                                        {
+                                            <td align="center">
+                                                <div align="center">
+                                                    <i class="fas fa-circle"
+                                                       style="opacity: @GetCellOpacity(gameState.PlayerPositions[i], board.GetCellIndex(row, col));
+                                                        color: @CellBackground[i];"></i>
+                                                </div>
+                                            </td>
+                                        }
+                                    </tr>
+                                    <tr>
+                                        @for (int i = 2; i < 4; i++)
+                                        {
+                                            <td align="center">
+                                                <div align="center">
+                                                    <i class="fas fa-circle"
+                                                       style="opacity: @GetCellOpacity(gameState.PlayerPositions[i], board.GetCellIndex(row, col));
+                                                        color: @CellBackground[i];"></i>
+                                                </div>
+                                            </td>
+                                        }
+                                    </tr>
+                                    </tbody>
+                                </table>
+                            </td>
+                        }
+                    }
+                </tr>
+            }
+            </tbody></table>
+    </Column>
+</Row>
+<hr>
+<Row><Column ColumnSize="ColumnSize.Is6">
+    <i class="fas fa-square" style="color: @CellBackground[5]; font-size: 150%;"></i> - the Player goes 3 cells BACK.
+    <br>
+    <i class="fas fa-square" style="color: @CellBackground[6]; font-size: 150%;"></i> - the Player goes 3 cells FORWARD.
+</Column></Row>
+
+@code {
+    private int diceValue;
+
+    private Task MoveAsync()
+    {
+        diceValue = GetRandomDigit();
+        var move = new DiceMove(MyPlayerIndex, diceValue);
```

Comment:
Yep, diceValue shouldn't be a part of DiceMove - it should be generated on server side when you process the move there.